Task1/weight.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# display the ideal weight using the formula <height> - 110
+class IdealWeight
+  SUBSTRACTOR = 110
+  EDGE = 0
+
+  def initialize
+    puts 'Enter your name: '
+    @name = gets.chomp.capitalize
+    puts 'Enter your height: '
+    @height = gets.chomp
+  end
+
+  def call
+    validate_parameters
+    detect_ideal_weight
+  end
+
+  private
+
+  def validate_parameters
+    return unless @height.to_i <= EDGE || @height.count('a-zA-Z').positive?
+
+    warn 'incorrect input for height!'
+    exit 1
+  end
+
+  def detect_ideal_weight
+    difference = @height.to_i - SUBSTRACTOR
```

Comment:
you can extract this to a method 

please read about Ruby Memoization
https://www.honeybadger.io/blog/ruby-rails-memoization/
https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/
https://thoughtbot.com/blog/ruby-memoization-and-alternatives

and use it in this method